internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/handler/ScanTask.java

```diff
@@ -670,38 +686,42 @@ public ScanStatus getScanStatus() throws IOException, XPathExpressionException,
-
-            String filePath = VeracodeScannerConfiguration.getInstance().getConfigProperty(
-                    VeracodeScannerConstants.VERACODE_OUTPUT_FOLDER_PATH) + File.separator + scanContext.getAppId();
-            FileUtil.saveReport(resultPdfDetailed, filePath + ScannerConstants.PDF_FILE_EXTENSION);
-            FileUtil.saveReport(resultXMLDetailed.getBytes(StandardCharsets.UTF_8.name()), filePath +
-                    VeracodeScannerConstants.SUMMARY + ScannerConstants.XML_FILE_EXTENSION);
-            FileUtil.saveReport(resultPdfSummary, filePath + ScannerConstants.PDF_FILE_EXTENSION);
-            FileUtil.saveReport(resultXMLSummary.getBytes(StandardCharsets.UTF_8.name()), filePath +
-                    VeracodeScannerConstants.SUMMARY + ScannerConstants.XML_FILE_EXTENSION);
-            FileUtil.saveReport(resultXMLThridParty, filePath + VeracodeScannerConstants.THIRD_PARTY
-                    + ScannerConstants.PDF_FILE_EXTENSION);
-            isReportPrinted = true;
-
-            String logMessage = "Scan reports are completed and downloaded to the location : " +
-                    VeracodeScannerConfiguration.getInstance().getConfigProperty(VeracodeScannerConstants.
-                            VERACODE_OUTPUT_FOLDER_PATH) + " for the application " + scanContext.getAppId();
-            log.info(logMessage);
-            CallbackUtil.persistScanLog(scanContext.getJobId(), logMessage, LogType.INFO);
-        } catch (IOException | ParserConfigurationException | SAXException | XPathExpressionException |
-                ScannerException e) {
-            String logMessage = "Error occured while downloading the sca reports for the application "
-                    + scanContext.getAppId();
-            log.error(logMessage);
-            CallbackUtil.persistScanLog(scanContext.getJobId(), logMessage, LogType.ERROR);
+        if (!Thread.currentThread().isInterrupted()) {
+            try {
+                ResultsAPIWrapper resultsAPIWrapper = VeracodeAPIUtil.getResultAPIWrapper();
+                String buildId = getBuildIDByAppId(scanContext.getAppId());
+                byte[] resultPdfDetailed = resultsAPIWrapper.detailedReportPdf(buildId);
+                String resultXMLDetailed = resultsAPIWrapper.detailedReport(buildId);
+                byte[] resultPdfSummary = resultsAPIWrapper.summaryReportPdf(buildId);
+                String resultXMLSummary = resultsAPIWrapper.summaryReport(buildId);
+                byte[] resultXMLThridParty = resultsAPIWrapper.thirdPartyReportPdf(buildId);
+
+                String filePath = VeracodeScannerConfiguration.getInstance().getConfigProperty(
+                        VeracodeScannerConstants.VERACODE_OUTPUT_FOLDER_PATH) + File.separator + scanContext.getAppId();
+                FileUtil.saveReport(resultPdfDetailed, filePath + ScannerConstants.PDF_FILE_EXTENSION);
+                FileUtil.saveReport(resultXMLDetailed.getBytes(StandardCharsets.UTF_8.name()), filePath +
```

Comment:
It's better to create the file names in separate variables and then pass here. Reader can know which file is getting saved in each instance, by looking at the variable name only.